src/application/engine/rendering/ClientMap.java

```diff
@@ -273,30 +287,37 @@ public void updateScores(HashMap<Integer, Double> scoreMap)
      * @param data
      *            Data about the new object.
      */
-    public void addGameObject(GameObjectDAO data) 
+    public void addGameObject(GameObjectDAO data)
     {
 
-        if(data.entityType != null)
+        if (data.entityType != null)
         {
-            if (data.ownerId == clientChar.getId()) 
+            if (data.ownerId == clientChar.getId())
             {
-                gameObjects.put(data.objectId, unassignedBullets.poll());
+                addGameObject(unassignedBullets.poll());
                 return;
             }
-            
-            gameObjects.put(data.objectId, EntityFactory.getEntity(data, data.entityType));
+
+            addGameObject(EntityFactory.getEntity(data, data.entityType));
+
         }
     }
 
+    private void addGameObject(GameObject go)
+    {
+        gameObjects.put(go.getId(), go);
+        go.addObserver(this);
+    }
+
     /**
-     * Removes a GameObject from the clientmap based on its ID.
+     * Destroys a GameObject from the clientmap based on its ID.
      * 
      * @param id
      *            The id of the object to be removed.
      */
-    public void removeGameObject(int id)
+    public void destroyGameObject(int id)
     {
-        gameObjects.remove(id);
+        gameObjects.get(id).destroy();
```

Comment:
we still want to remove the object here and not let that be up to the observable to do so